.github/workflows/windows-ci.yml

```diff
@@ -0,0 +1,230 @@
+    tags:
+      - windows-*
+
+  pull_request:
+    branches:
+      - '*'
+
+env:
+  SCCACHE_GHA_ENABLED: "true"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  # INFO: We are cancelling the concurrency group if the change is on PR. For workflow dispatch, this will not work.
+  cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
+
+permissions:
+  actions: read           # needed for CodeQL
+  contents: read          # needed for CodeQL
+  security-events: write  # needed for CodeQL
+
+jobs:
+  pre-job:
+    runs-on: windows-2022
+    outputs:
+      should_skip: ${{ github.event_name != 'workflow_dispatch' && steps.changed-files.outputs.windows_any_modified != 'true' }}
+    steps:
+      - run: |
+          git config --system core.longpaths true
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Get all Windows files that have changed
+        if: github.event_name != 'workflow_dispatch'
+        id: changed-files
+        uses: tj-actions/changed-files@v45
+        with:
+          files_yaml_from_source_file: .github/changed-files.yml
+
+      - name: List changed files
+        if: steps.changed-files.outputs.windows_any_modified == 'true'  
+        shell: bash
+        run: |
+          echo "Changed file(s): ${{ steps.changed-files.outputs.windows_all_changed_files }}"
+
+  windows-build-and-test:
+    if: needs.pre-job.outputs.should_skip != 'true'
+    needs: pre-job
+    strategy:
+      matrix:
+        renderer: [opengl, egl, vulkan, osmesa]
+        rendering_mode: [legacy, drawable]
+        exclude:
+          - renderer: egl
+            rendering_mode: drawable
+          - renderer: vulkan
+            rendering_mode: legacy
+    runs-on: windows-2022
+    steps:
+      - run: |
+          git config --system core.longpaths true
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+          fetch-depth: 0
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v3
+        with:
+          languages: cpp
+
+      - uses: ilammy/msvc-dev-cmd@v1
+
+      - name: Export GitHub Actions cache environment variables
+        uses: actions/github-script@v7
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+
+      - uses: mozilla-actions/sccache-action@v0.0.6
+
+      - name: Initialize sccache
+        run: |
+          sccache --start-server
+          sccache --zero-stats
+
+      - name: Configure MapLibre Native Core
+        env:
+          CI: 1
+          VCPKG_BINARY_SOURCES: "clear;x-gha,readwrite"
+          VCPKG_KEEP_ENV_VARS: CMAKE_CXX_COMPILER_LAUNCHER;CMAKE_C_COMPILER_LAUNCHER
+          CMAKE_C_COMPILER_LAUNCHER: sccache
+          CMAKE_CXX_COMPILER_LAUNCHER: sccache
+        shell: bash
+        run: |
```

Comment:
Shellcheck detects an issue with this script:

```
Argument mixes string and array. Use * or separate argument
```

Reply:
Since this is quite a large script, maybe it can be extracted in a file so it can be easily ran.

Reply:
You are right. Done!

Reply:
Now Shellcheck tests passed =D!